libsh/docs/libTest.man
.\" Automatically generated by Pod::Man 2.27 (Pod::Simple 3.28)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "LIBTEST.SH 1"
.TH LIBTEST.SH 1 "2017-02-11" "perl v5.18.2" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "Name"
.IX Header "Name"
libTest.sh
.SH "Description"
.IX Header "Description"
Test library. aide a\*` tester le comportement d un script dans un environnement maitrise.
.SH "Dependencies"
.IX Header "Dependencies"
libError.sh
.SH "Functions"
.IX Header "Functions"
.SS "assert_return_code"
.IX Subsection "assert_return_code"
assert_return_code \*(L"\s-1COMMAND\*(R" \*(L"EXPECTED_RETURN_CODE\*(R"\s0
.PP
.Vb 2
\& RETURN 0 if EXPECTED_RETURN_CODE == RETURN_CODE
\& RETURN 1 othewise
.Ve
.PP
in both case there are a message of type
.PP
\&\fBhostname \- date time \- program \-(function_tested) return_code;expected_return_code state\fR
.PP
tel que :
.PP
myhost \- 0211 15:39 \- :  ./test.sh \-(ls) code retour:0 ; attendu:1  \s-1KO \s0
.SS "assert_return_err"
.IX Subsection "assert_return_err"
\&\s-1TODO :\s0 verify stderr
.SS "assert_return_message"
.IX Subsection "assert_return_message"
\&\s-1TODO :\s0 verify stdout
.SH "Author"
.IX Header "Author"
Grillon


libsh/libTest.sh
#!/usr/bin/env bash

<<=cut

=pod

=encoding utf-8

=head1 Name

libTest.sh

=head1 Description

Test library. aide à tester le comportement d un script dans un environnement maitrise.

=head1 Dependencies

libError.sh

=head1 Functions

=head2 assert_return_code

assert_return_code "COMMAND" "EXPECTED_RETURN_CODE"

 RETURN 0 if EXPECTED_RETURN_CODE == RETURN_CODE
 RETURN 1 othewise

in both case there are a message of type

B<hostname - date time - program -(function_tested) return_code;expected_return_code state>

tel que :

myhost - 0211 15:39 - :  ./test.sh -(ls) code retour:0 ; attendu:1  KO 

=head2 assert_return_err

TODO : verify stderr

=head2 assert_return_message

TODO : verify stdout

=head1 Author

Grillon

=cut

#dependancy
if [ ! "$libError" ];then
	echo "lib de gestion d'erreur manquante" >&2
	exit 1
fi

function assert_return_code {
eval $HEADER
typeset cmd=$1
typeset code_expected=$2
typeset code_obtained
typeset result
$cmd &>/dev/null
code_obtained=$?
if [ "${code_obtained}" -eq "${code_expected}" ];then
	result=$OK
else
	result=$KO
fi
erreur "${result}" "$mf code retour:${code_obtained} ; attendu:${code_expected}" $ECONT
}
function assert_return_err {
echo "TODO"
}
function assert_return_message {
echo "TODO"
}
